scheduler/src/cook/mesos/util.clj

```diff
@@ -198,38 +199,73 @@
      get-pending-jobs-duration
      (get-pending-job-ents* unfiltered-db true))))
 
+(defn generate-intervals
+  "Generates a list of intervals between start and end as pairs
+   [interval-start interval-end]. The union of the intervals is
+   inclusive of both start and end
+
+   Parameters:
+   ----------
+   start : clj-time/datetime
+   end : clj-time/datetime
+   period-like : clj-time/period
+
+   Returns:
+   --------
+   list of pairs, [interval-start interval-end]"
+  ([start end period-like]
+   (->> (conj (vec (periodic-seq start end period-like)) end)
+        (partition 2 1))))
+
+(defn accumulate-n-from-batches
+  "Given a seq of collections, `batches`, take `n` elements from the collections,
+   starting with the first and realizing subsequent batches once previous batches
+   are consumed."
+  [n batches]
+  (loop [coll []
+         [batch & batches] batches]
+    (let [needed-elements (- n (count coll))
```

Comment:
`needed-elements` can become a loop variable and save the call on `(count coll)`.